### ISSUE TITLE
magento 2.3 Design Configuration > Global page keep loading

### ISSUE THREAD

**Issue by andy17612** [magento/magento2#19549]
### Preconditions (*)

1. magento 2.3.0
2. php 71
3. mysql 5.7

### Steps to reproduce (*)

1. Open Admin Panel.
2. Chose "Content"
3. Chose "Design" ==> "Configuration"
4. After Design Configuration page loaded, then choose "Global" and click "Edit"

### Expected result (*)

1. Ussually on Global Design Configuration, we can upload Logo, email logo, and icon for homepage.

### Actual result (*)

1. It is freeze nothing happened. Only loading never ending. Even you refresh the page cannot succesfully loaded to show you some dialog for header, email or icon logo. Like this screenshot.
2. same with https://github.com/magento/magento2/issues/19486 but i think is not theme issue.
3. **Main Website and Main Website Store are working, just Global are not working**

This is Global page message. please help, thank you!

```
file-uploader.min.js:1 Uncaught ReferenceError: Base64 is not defined
    at UiClass.processFile (file-uploader.min.js:1)
    at Array.map (<anonymous>)
    at UiClass.setInitialValue (file-uploader.min.js:1)
    at UiClass.initialize (abstract.min.js:1)
    at UiClass.initialize (wrapper.min.js:3)
    at new UiClass (class.min.js:3)
    at Object.initComponent (layout.min.js:8)
    at fire (jquery.min.js:212)
    at Object.fireWith [as resolveWith] (jquery.min.js:218)
    at Object.deferred.(/index.php/admin/theme/design_config/edit/scope/default/key/3d098cdba6b78b5b34afeb20d3e1f3d052972a2b3b29bc3d63761/anonymous function) (http://www.domain.com/pub/static/version154399332/adminhtml/Magento/backend/en_US/jquery.min.js:220:56)
```

**Comment by magento-engcom-team** [comment]
Hi @andy17612. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@andy17612 do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by andy17612** [comment]
@magento-engcom-team  from 2.2.6 upgrade to 2.3 after getting this problem, is not a new 2.3.

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by andy17612** [comment]
@magento-engcom-team Main Website and Main Website Store are working, just Global are not working

**Comment by ghost** [comment]
Hi @andy17612 thank you for you report, this issue has already fixed in 2.3-develop branch and will be available on 2.3.1 release.

**Comment by jurpro** [comment]
@engcom-backlog-nazar how to fix? we need change logo from developer theme logo to my own logo. Meanwhile, my web still use dev logo. Is there any change file? Where I will find the fix?

**Comment by ghost** [comment]
@andy17612  https://github.com/magento/magento2/pull/18851

**Comment by boostify-pro** [comment]
Having same problem here, any solution please

**Comment by andy17612** [comment]
@SoCix file in there,but i'm testing too  #18851 https://github.com/magento/magento2/commit/c96b8affa550f55359c3302226ad4d095e5d933b#diff-5a423cde3ac7f2aed85a0994b5597827L54

**Comment by andy17612** [comment]
@engcom-backlog-nazar #18851 https://github.com/magento/magento2/commit/c96b8affa550f55359c3302226ad4d095e5d933b#diff-5a423cde3ac7f2aed85a0994b5597827L54  I tried this file but when you Transactional email >upload logo > not showing.

**Comment by laquo** [comment]
Same problem here. Tried c96b8af#diff-5a423cde3ac7f2aed85a0994b5597827L54 
After that i've also tried to select logo from gallery. It appears in the uploader block, but anyway cause an error "header_logo_src does not contain field 'file'" after i tried to save it

**Comment by woshka** [comment]
same issue
how to fix this?

**Comment by mikeguy** [comment]
Same issue here. No fix for this yet?

**Comment by afavata** [comment]
Just upgraded to 2.3 and Design Configuration - Global or View page do keep loading infinitively. Website Page does work properly.
Is there a patch to quickly resolve this issue???
Thanks

**Comment by michelecanal** [comment]
Look at andy17612 solution

**Comment by boostify-pro** [comment]
Open those files

```
> /vendor/magento/module-email/view/adminhtml/ui_component/design_config_form.xml
> /vendor/magento/module-catalog/view/adminhtml/ui_component/design_config_form.xml
> /vendor/magento/module-backend/view/adminhtml/ui_component/design_config_form.xml
> /vendor/magento/module-swatches/view/adminhtml/ui_component/design_config_form.xml
> /vendor/magento/module-theme/view/adminhtml/ui_component/design_config_form.xml
```

search for

`formElement="fileUploader"`

replace with

`formElement="imageUploader"
`
in all files

clear cache and reload.

**Comment by mikeguy** [comment]
Actually now I can't access the Design Configuration - Global View page as well just like @afavata. It sits there spinning that icon like something's loading but it's just stuck.

@SoCix didn't try your suggestion as it seems the issue is different, page just doesn't load^
@michelecanal same, seems I have a different issue really.

I can't find something relating to this.
![screenshot_26](https://user-images.githubusercontent.com/12158143/50008728-12711680-ff83-11e8-8d38-c3c5725933e8.png)

**Comment by measureddesigns** [comment]
Having exactly the same issue on 2.3.0, so can't change the theme. Spinner never goes away. Base64 is not defined -  file-uploader.js

**Comment by mikeguy** [comment]
@engcom-backlog-nazar seems this issue is trending.

**Comment by ghost** [comment]
@mikeguy this will be available on 2.3.1 release. and already fixed in 2.3-dev.

**Comment by mikeguy** [comment]
@engcom-backlog-nazar thank you for super fast response.
I take it no workaround until then?

**Comment by ghost** [comment]
@andy17612 this bug not fixed yet ( -> https://github.com/magento/magento2/issues/19581

**Comment by ghost** [comment]
@mikeguy seems like no :(

**Comment by boostify-pro** [comment]
@mikeguy I had the same problem and this was the right solution 
Once you make those changes you’ll be able to access the page again 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by webcitz** [comment]
@SoCix is correct, but you might find and need to update 3 instances of fileUploader in the 2nd file and 2 instances of fileUploader in the 5th file.

**Comment by evagabond** [comment]
@SoCix worked for me. Thanks mate!

**Comment by whitworthprints** [comment]
this worked for me - but setup:di:compile no longer returns cleanly 

		Invalid ExtensionInterface for nonexistent class Interface in file /var/www/vhosts/DOMAIN/app/code/Magento/TestModuleExtensionAttributes/Api/Data/FakeExtensibleTwoInterface.php

**Comment by woshka** [comment]
delete these directories
var/cache 
var/page_cache
var/view_preprocessed
then di compile

**Comment by whitworthprints** [comment]
Same :(

[magento2@server var]$ rm -rf cache page_cache view_preprocessed 
[magento2@server var]$ cd ..
[magento2@server DOMAIN]$ bin/magento setup:di:compile
Compilation was started.
Interception cache generation... 6/7 [========================>---]  85% 46 secs 405.0 MiBErrors during configuration scanning:
	int
		Invalid ExtensionInterface for nonexistent class Interface in file /var/www/vhosts/DOMAIN/app/code/Magento/TestModuleExtensionAttributes/Api/Data/FakeExtensibleTwoInterface.php
Total Errors Count: 1

In Log.php line 92:
                            
  Error during compilation  
                            

setup:di:compile

**Comment by woshka** [comment]
there has been some Unit test Php codes that are generated in your app/code/Magento in
/var/www/vhosts/DOMAIN/app/code/Magento/
get rid of all of them safly
backup the directory before that
compile again after removal
thx

**Comment by whitworthprints** [comment]
Thanks that enabled me to run a successful compile, and put it back to production mode, but sadly the issue i was having is back again, which this appeared to fix (adding products to basket with a custom picture is back again - in both production and developer mode), it just seems to refresh the screen when you click add to cart and nothing gets added :(
I may have to just wait till 2.3.1 is released and see how that goes :(

**Comment by woshka** [comment]
there has been no problem of such there seems to be some other issues you may facing
please test your website Client side errors by pressing Shift + CTRL + I and see if there are any errors

**Comment by whitworthprints** [comment]
Im seeing this in the console after the add to cart refreshs screen?

jquery.js:10208 [Deprecation] Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user's experience. For more help, check https://xhr.spec.whatwg.org/.

**Comment by woshka** [comment]
then it is a client side issue that does not has any thing to do with magento 2.3.0
you should fix your JS issue

**Comment by whitworthprints** [comment]
Version 71.0.3578.98 (Official Build) (64-bit)
the latest version of chrome, no updates available - so how would one look into fixing something to do with a browser?

**Comment by simonmaass** [comment]
@SoCix i did as suggested... i get the following JS error:

file-uploader.js:173 Uncaught ReferenceError: Base64 is not defined
    at UiClass.processFile (file-uploader.js:173)
    at Array.map (<anonymous>)
    at UiClass.setInitialValue (file-uploader.js:78)
    at UiClass.initialize (abstract.js:77)
    at UiClass.initialize (wrapper.js:109)
    at new UiClass (class.js:49)
    at Object.initComponent (layout.js:137)
    at fire (jquery.js:3232)
    at Object.fireWith [as resolveWith] (jquery.js:3362)
    at Object.deferred.(/admin/theme/design_config/edit/scope/stores/scope_id/1/key/28725cf24260870dc3061e4cdc1c1b3b34f5405c043832ee44fa9fbc5cace383/anonymous function) (https://d3r70qgxuk5xcb.cloudfront.net/adminhtml/meevo/admin/en_US/jquery.js:3461:37)

**Comment by ghost** [comment]
@simonmaass https://github.com/magento/magento2/pull/19988 this will fix your issue.

**Comment by Zyles** [comment]
Link to solution? Need a workaround right now.

**Comment by Nazar65** [comment]
@Zyles try this -> https://github.com/magento/magento2/pull/18851/files

**Comment by oliverf72** [comment]
@Nazar65 there is no folder /app/code/Magento... ?!

**Comment by Nazar65** [comment]
@oliverf72 if you have a composer installation they must be like that `vendor/magento/module-theme/view/adminhtml/ui_component/design_config_form.xml`

**Comment by oliverf72** [comment]
@Nazar65 perfect, solved my issue...thanks a lot...

**Comment by R-K-Shrivastaw** [comment]
I am having the same error in my development magento. Cant upload the logo. If anyone knows how to fix it, please help me.

**Comment by Prasad2706** [comment]
> imageUploader

@SoCix
solution not work for me :(
magento version 2.3.0

**Comment by simonmaass** [comment]
this is fixed in 2.3.1...

**Comment by Prasad2706** [comment]
> this is fixed in 2.3.1...

but we need to work on 2.3 cant upgrade to 2.3.1

**Comment by simonmaass** [comment]
this is the patch we applied for 2.3.0
[gh_18851_design_icon.zip](https://github.com/magento/magento2/files/3072628/gh_18851_design_icon.zip)

**Comment by bheindl** [comment]
I'm not sure if changing input type from fileUploader to imageUploader solves the basic issue. When I look at the error message and the code, the problem is, that "Base64"-Method from lib/web/mage/adminhtml/tools.js is not available in global scope.
I can reproduce this issue, when I activate 'dev/js/enable_js_bundling' and 'dev/js/minify_files' and then compile the static content. Tools.js is somehow compiled into a bundle.js but the "Base64" function is not available in global scope anymore (and any other function from tools.js is also missing in global scope)

**Comment by Nazar65** [comment]
@bheindl  https://github.com/magento/magento2/pull/19988

**Comment by visahardik** [comment]
> Open those files
> 
> ```
> > /vendor/magento/module-email/view/adminhtml/ui_component/design_config_form.xml
> > /vendor/magento/module-catalog/view/adminhtml/ui_component/design_config_form.xml
> > /vendor/magento/module-backend/view/adminhtml/ui_component/design_config_form.xml
> > /vendor/magento/module-swatches/view/adminhtml/ui_component/design_config_form.xml
> > /vendor/magento/module-theme/view/adminhtml/ui_component/design_config_form.xml
> ```
> 
> search for
> 
> `formElement="fileUploader"`
> 
> replace with
> 
> `formElement="imageUploader" `
> in all files
> 
> clear cache and reload.

vendor/magento/module-email/view/adminhtml/ui_component/design_config_form.xml


you have to change this file.

**Comment by bassmo** [comment]
> there has been some Unit test Php codes that are generated in your app/code/Magento in
> /var/www/vhosts/DOMAIN/app/code/Magento/
> get rid of all of them safly
> backup the directory before that
> compile again after removal
> thx

What does "get rid of them safely" mean? Removing the entire directory causes a whole load of errors.

**Comment by LiamKarlMitchell** [comment]
So I ran this as mentioned above on a Magento 2.3.0 (that was upgraded from a prior one somehow) appears to fix it.

```bash
sed -i 's/formElement="fileUploader"/formElement="imageUploader"/g' vendor/magento/module-email/view/adminhtml/ui_component/design_config_form.xml
sed -i 's/formElement="fileUploader"/formElement="imageUploader"/g' vendor/magento/module-catalog/view/adminhtml/ui_component/design_config_form.xml
sed -i 's/formElement="fileUploader"/formElement="imageUploader"/g' vendor/magento/module-backend/view/adminhtml/ui_component/design_config_form.xml
sed -i 's/formElement="fileUploader"/formElement="imageUploader"/g' vendor/magento/module-swatches/view/adminhtml/ui_component/design_config_form.xml
sed -i 's/formElement="fileUploader"/formElement="imageUploader"/g' vendor/magento/module-theme/view/adminhtml/ui_component/design_config_form.xml
bin/magento cache:flush
```
